packages/frontend/src/pages/Dashboard.tsx

```diff
@@ -0,0 +1,228 @@
+import { FC, useEffect, useState } from 'react';
+import { Skeleton, Table, formatTokenAmount } from '@sifi/shared-ui';
+import { Token } from '@sifi/sdk';
+import { Button } from 'src/components/Button';
+import { PartnerTokensByChain, TokenOfPartner } from 'src/hooks/usePartnerTokens';
+import { getChainById, getChainIcon } from 'src/utils/chains';
+import { useAccount, useNetwork, useSwitchNetwork } from 'wagmi';
+import { useFetchTokens } from 'src/hooks/useFetchTokens';
+import { usePartnerWithdraw } from 'src/hooks/usePartnerWithdraw';
+import { ConnectWallet } from 'src/components/ConnectWallet/ConnectWallet';
+import { firstAndLast, getEvmTxUrl } from 'src/utils';
+import { ETH_CONTRACT_ADDRESS, ETH_ZERO_ADDRESS } from 'src/constants';
+import { usePartnerData } from 'src/hooks/usePartnerData';
+
+const extractTokenAddress = (id: string): string => {
+  // The id is a concatenation of two Ethereum addresses,
+  // the token address is last 40 characters of the id.
+  return '0x' + id.slice(-40);
+};
+
+type PartnerTokensTableProps = {
+  partnerTokensByChain: PartnerTokensByChain;
+};
+
+type TokenRowProps = {
+  token: TokenOfPartner;
+  chainId: string;
+  handleWithdraw: (id: string, chainId: number) => void;
+  withdrawnTokens: { [key: string]: string };
+  tokensByChainIdAndAddress: Record<string, Token>;
+};
+
+const TokenRow: FC<TokenRowProps> = ({
+  token,
+  chainId,
+  handleWithdraw,
+  withdrawnTokens,
+  tokensByChainIdAndAddress,
+}) => {
+  let tokenAddress = extractTokenAddress(token.id);
+  if (tokenAddress === ETH_ZERO_ADDRESS) {
+    tokenAddress = ETH_CONTRACT_ADDRESS.toLowerCase();
+  }
+  const tokenKey = `${chainId}-${tokenAddress}`;
+  const tokenDetails = tokensByChainIdAndAddress[tokenKey];
+  const showWithdrawalButton = Number(token.balanceDecimal) > 0 && !withdrawnTokens[token.id];
+  const lastWithdrwalHash = withdrawnTokens[token.id] || token.modifiedAtTransaction;
+  const lastWithdrwalUrl = getEvmTxUrl(
+    getChainById(Number(chainId)),
+    withdrawnTokens[token.id] || token.modifiedAtTransaction
+  );
+  const hasBalance = Number(token.balanceDecimal) > 0;
+
+  return (
+    <Table.Row className="overflow-y-auto max-w-xs m-auto sm:max-w-none my-2" key={token.id}>
+      <Table.Cell className="mb-2 w-full sm:mb-0">
```

Comment:
I think it would be better to use `Table.Cell` twice for semantic HTML, for both columns, instead of as a wrapper with a grid inside.

Reply:
[dd25c2f](https://github.com/sifiorg/sifi/pull/401/commits/dd25c2fc2d5a985966f371059c9003b9135fc657)